### PR TITLE
[KOGITO-5710] - Add support for Kogito Quarkus runtime native image

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>
See: https://issues.redhat.com/browse/KOGITO-5710

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by spolti** [comment]
@radtriste  do you know why the examples is failing to get built?

File: rhpam-kogito-imagestream.yaml [hunk 1]
```diff
@@ -27,18 +27,39 @@ items:
...
+      annotations:
+        openshift.io/display-name: RHPAM Runtime image for Kogito based on Quarkus native image
+        openshift.io/provider-display-name: Kie Group.
+    spec:
+      tags:
+        - name: '7.11.0'
+          annotations:
+            description: RHPAM Runtime image for Kogito based on Quarkus native image
+            iconClass: icon-jbpm
+            tags: rhpam-kogito,runtime,kogito,quarkus
```
Comment (radtriste) [inline rhpam-kogito-imagestream.yaml]:
```suggestion
            tags: rhpam-kogito,runtime,kogito,quarkus,native
```
?

File: rhpam-kogito-runtime-native-rhel8-overrides.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+from: "registry.redhat.io/ubi8/ubi-minimal:latest"
+description: "RHPAM Runtime image for Kogito based on Quarkus native image"
+
+labels:
+  - name: "com.redhat.component"
+    value: "rhpam-7-kogito-runtime-native-rhel8-container"
+  - name: "io.k8s.description"
+    value: "RHPAM Runtime image for Kogito based on Quarkus native image"
+  - name: "io.k8s.display-name"
+    value: "RHPAM Kogito based on Quarkus native image"
```
Comment (radtriste) [inline rhpam-kogito-runtime-native-rhel8-overrides.yaml]:
```suggestion
    value: "Red Hat build of Kogito runtime based on Quarkus native image"
```

File: rhpam-kogito-runtime-native-rhel8-overrides.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  - name: "io.openshift.tags"
+    value: "builder,runtime,kogito,quarkus,native"
```
Comment (radtriste) [inline rhpam-kogito-runtime-native-rhel8-overrides.yaml]:
```suggestion
    value: "rhpam-kogito,runtime,kogito,quarkus,native"
```

File: rhpam-kogito-runtime-native-rhel8-overrides.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    value: "/home/kogito/bin"
+
+envs:
+  - name: "SCRIPT_DEBUG"
+    example: "true"
+    description: "If set to true, ensures that the bash scripts are executed with the -x option, printing the commands and their arguments as they are executed. Also debug JVM initialization."
+  - name: JAVA_OPTIONS
+    example: "-Dquarkus.log.level=DEBUG"
+    description: JVM options passed to the Java command.
+  - name: BINARY_BUILD
```
Comment (radtriste) [inline rhpam-kogito-runtime-native-rhel8-overrides.yaml:27]: I guess `BINARY_BUILD` should always be `true` for now ?

File: rhpam-kogito-runtime-native-rhel8-overrides.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+packages:
+  content_sets_file: content_sets.yaml
+  install:
+    - glibc
+    - jq
+
+osbs:
+  configuration:
+    container_file: container.yaml
+  extra_dir: osbs-extra/rhpam-kogito-runtime-jvm-rhel8
```
Comment (radtriste) [inline rhpam-kogito-runtime-native-rhel8-overrides.yaml]:
```suggestion
  extra_dir: osbs-extra/rhpam-kogito-runtime-native-rhel8
```

File: tests/features/kogito-common-runtime-native.feature [hunk 6]
```diff
@@ -0,0 +1,21 @@
+@quay.io/kiegroup/kogito-runtime-native
+@rhpam-7/rhpam-kogito-runtime-native-rhel8
+Feature: Kogito-runtime-native feature.
+
+  Scenario: Verify if the binary build is finished as expected and if it is listening on the expected port with quarkus native
```
Comment (radtriste) [inline tests/features/kogito-common-runtime-native.feature:5]: Could you also delete this scenario from `tests/features/kogito-runtime-native.feature` file ?

**Comment by radtriste** [comment]
Seems that the `nightly-main` branch has not been updated in the past days due to test failures on examples with onboarding.
It should be solved today with https://github.com/kiegroup/kogito-examples/pull/849

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Could the error be related to https://kie.zulipchat.com/#narrow/stream/232676-kogito/topic/jobs.20service.20pg.20exits.20with.20no.20error ?

File: rhpam-kogito-runtime-native-rhel8-overrides.yaml (hunk 5 above)
Comment (spolti) [reply rhpam-kogito-runtime-native-rhel8-overrides.yaml]: oops, too many copy and paste :)

File: rhpam-kogito-runtime-native-rhel8-overrides.yaml (hunk 4 above)
Comment (spolti) [reply rhpam-kogito-runtime-native-rhel8-overrides.yaml:27]: it makes sense.

File: tests/features/kogito-common-runtime-native.feature (hunk 6 above)
Comment (radtriste) [reply tests/features/kogito-common-runtime-native.feature:5]: Forgot that one ?

File: osbs-extra/rhpam-kogito-runtime-native-rhel8/gating.yaml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+--- !Policy
+id: "cvp-rhpam-kogito-runtime-jvm"
```
Comment (radtriste) [inline osbs-extra/rhpam-kogito-runtime-native-rhel8/gating.yaml]:
```suggestion
id: "cvp-rhpam-kogito-runtime-native"
```

**Review by radtriste — CHANGES_REQUESTED** [review]

**Review by sutaakar — APPROVED** [review]

File: tests/features/kogito-common-runtime-native.feature (hunk 6 above)
Comment (spolti) [reply tests/features/kogito-common-runtime-native.feature:5]: what do you mean?
This test is valid for both images.

Comment (radtriste) [reply tests/features/kogito-common-runtime-native.feature:5]: yes it is.
But now you have 2 scenarios with same name and content: `Verify if the binary build is finished as expected and if it is listening on the expected port with quarkus native`
As you moved that one to common file, can you remove the scenario from the original file, aka `tests/features/kogito-runtime-native.feature` as it makes no sense to test it twice ?

Comment (radtriste) [reply tests/features/kogito-common-runtime-native.feature:5]: See https://github.com/kiegroup/kogito-images/pull/671/commits/9a46bd8daeb34cb417441d91050155a8e6fafff2 that I added to PR

**Review by radtriste — APPROVED** [review]
you can merge if https://github.com/kiegroup/kogito-images/pull/671/commits/9a46bd8daeb34cb417441d91050155a8e6fafff2 is ok for you

**Pull request merged**